src/main/scala/io/iohk/ethereum/vm/DataWord.scala

```diff
@@ -79,10 +88,56 @@ class DataWord private (private val n: BigInt) extends Ordered[DataWord] {
 
   def *(that: DataWord): DataWord = DataWord(this.n * that.n)
 
-  def /(that: DataWord): DataWord = DataWord(this.n / that.n)
+  def /(that: DataWord): DataWord = zeroCheck(that) { DataWord(this.n / that.n) }
+
+  def sdiv(that: DataWord): DataWord = zeroCheck(that) {
+    DataWord(toUnsignedBigInt(this.signedN / that.signedN))
+  }
+
+  def mod(that: DataWord): DataWord = zeroCheck(that) { DataWord(this.n mod that.n) }
+
+  def smod(that: DataWord): DataWord = zeroCheck(that) {
+    DataWord(this.signedN.signum * (this.signedN.abs mod that.signedN.abs))
+  }
+
+  def addmod(that: DataWord, modulus: DataWord): DataWord = zeroCheck(modulus) {
+    DataWord((this.n + that.n) mod modulus.n)
+  }
+
+  def mulmod(that: DataWord, modulus: DataWord): DataWord = zeroCheck(modulus) {
+    DataWord((this.n * that.n) mod modulus.n)
+  }
 
   def **(that: DataWord): DataWord = DataWord(this.n.modPow(that.n, Modulus))
 
+  def signExtend(that: DataWord): DataWord = {
```

Comment:
Regarding @rtkaczyk said about the `.toByte` conversion being wrong in `getByte`, isn't it also the case here? Geth also